server/bennu-core/src/main/java/org/fenixedu/bennu/core/rest/JsonBodyReaderWriter.java

```diff
@@ -50,7 +50,8 @@ public JsonBodyReaderWriter() {
 
     @Override
     public boolean isReadable(Class<?> type, Type genericType, Annotation[] annotations, MediaType mediaType) {
-        return (JsonElement.class.isAssignableFrom(type) || JsonElement.class.isAssignableFrom((Class<?>) genericType));
+        // ignore media type parameters
+        return MediaType.APPLICATION_JSON_TYPE.equals(new MediaType(mediaType.getType(), mediaType.getSubtype()));
```

Comment:
Doesn't this blow up if the invoked method has more parameters than just the `body` one?